Chromium Chronicle 20 (#682)
GoogleChrome/developer.chrome.com: site/en/blog/chromium-chronicle-20/index.md

---

> title: "The Chromium Chronicle #20: Benchmarking Test Harnesses"

Reviewer: Is "#20" consistent with the other editions? (If this were the first edition, I recommend "1" instead of "#1".)

Author: Yep, it's consistent.

description: >
  Adding benchmarks is an easy way to prevent performance regressions and

>   help improve performance over time.

Reviewer: ```suggestion
  help improve performance.
```

layout: 'layouts/blog-post.njk'
date: 2021-04-30
hero: 'image/0g2WvpbGRGdVs0aAPc6ObG7gkud2/hgu6uTktp2ipmuODZZhP.jpg'
alt: >
  Chromium Chronicle image
tags:
  - chromium-chronicle
---

**Episode 20:** by John Chen in Bellevue, WA (April, 2021)<br>
[Previous episodes](/tags/chromium-chronicle/)

> Speed is one of Chrome's four core principles. **Adding benchmarks is an

Reviewer: ```suggestion
Speed is one of Chrome's four core principles and has several benefits.

**Adding benchmarks is an
```

easy way to prevent performance regressions and help improve performance
over time.** Good Benchmarks have a fast iteration cycle, can catch
performance regressions much earlier than UMA, and are great for measuring

> new feature's performance.

Reviewer: ```suggestion
a new feature's performance.
```

Reviewer: This one is a grammar mistake and has to be fixed.

> **Benchmarks run regularly in the lab.** When a regression is found, bisect

Reviewer: 'Bisect' is a command or system? This should either be:

* 'the Bisect command automatically finds...' (or something similar)

or

* 'Bisecting automatically finds...'

Author: Updated with link to bisect

> automatically finds the culprit CL and a bug is assigned to the CL owner.

Reviewer: ```suggestion
automatically finds the culprit CL and assigns a bug to the CL owner.
```

**Chrome benchmarks combine sequences of web page interactions (called
stories) with performance measurements.** Similar cases are grouped into
benchmark harnesses. New benchmarks usually fit into one of the
[existing harnesses](http://bit.ly/chrome-benchmark-harnesses):

{% Columns %}

{% Column %}

* System Health
* Loading
* Memory
* Rendering
* Power
* Startup

{% endColumn %}
{% Column %}

* V8 Runtime
* Media
* WebRTC
* Press
* Blink Perf

{% endColumn %}
{% endColumns %}

**The Telemetry framework replays recorded stories to simulate user
interactions with Chrome while collecting traces that record Chrome
activities.** After the stories finish, the framework runs various performance
metrics to analyze the traces and calculates performance results.

You can cover most new performance test cases in Chrome by adding a new story,
using an existing metric within one of the existing harnesses. You can also
collect additional traces and add more metrics to existing benchmarks or pass
additional flags to the browser.

> Use Blink Perf for one off cases that don't fit into other harnesses. In

Reviewer 2: Nit: one off --> one-off to stay consistent with the other usage of "one-off" spelling.

> Use Blink Perf for one off cases that don't fit into other harnesses. In

Reviewer: ```suggestion
Use Blink Perf for one-off cases that don't fit into other harnesses. In
```

Blink Perf, you can measure trace events on one-off pages.

> **Keep your benchmark stories simple** and only include the minimal set of

Reviewer: You've transitioned from a list of benefits to giving advice, but I didn't get text that introduces the shift.

interactions needed to complete your scenario. If a test case is complex, it

> may be hard to automate or be flaky.

Reviewer: ```suggestion
may be hard to automate or it may be flaky.
```
